src/client/Microsoft.Identity.Client.Broker/RuntimeBroker.cs

```diff
@@ -435,26 +477,59 @@ public bool IsBrokerInstalledAndInvokable(AuthorityType authorityType)
         {
             if (!DesktopOsHelper.IsWin10OrServerEquivalent())
             {
-                _logger.Warning("[WAM Broker] Not a supported operating system. WAM broker is not available. ");
+                _logger?.Warning("[WAM Broker] Not a supported operating system. WAM broker is not available. ");
                 return false;
             }
 
             // WAM does not work on pure ADFS environments
             if (authorityType == AuthorityType.Adfs)
             {
-                _logger.Warning("[WAM Broker] WAM does not work in pure ADFS environments. Falling back to browser for an ADFS authority unless Proof-of-Possession is configured. ");
+                _logger?.Warning("[WAM Broker] WAM does not work in pure ADFS environments. Falling back to browser for an ADFS authority unless Proof-of-Possession is configured. ");
                 return false;
             }
 
             if (s_lazyCore.Value == null)
             {
-                _logger.Info("[WAM Broker] MsalRuntime initialization failed. See https://aka.ms/msal-net-wam#wam-limitations");
-                _logger.InfoPii(s_initException);
+                _logger?.Info("[WAM Broker] MsalRuntime initialization failed. See https://aka.ms/msal-net-wam#wam-limitations");
+                _logger?.InfoPii(s_initException);
                 return false;
             }
 
-            _logger.Verbose($"[WAM Broker] MsalRuntime initialization successful.");
+            _logger?.Verbose($"[WAM Broker] MsalRuntime initialization successful.");
             return true;
         }
+
+        internal class LogEventWrapper : IDisposable
+        {
+            private bool _disposedValue;
+            RuntimeBroker _broker;
+
+            public LogEventWrapper(RuntimeBroker broker)
+            {
+                _broker = broker;
+                s_lazyCore.Value.LogEvent += _broker.LogEventRaised;
```

Comment:
I think it will be difficult to repro this manually. Maybe using blocking tasks, like when we test WithCacheSynchronization.

But it could happen if request 1 is disposing and unsubscribed the event handler, while request 2 is in the middle of doing work.

I forget, is he event collection a set or a list? Would calling `+= _broker.LogEventRaised;` multiple times, add multiple delegates or just 1?

Reply:
> forget, is he event collection a set or a list? W
This is a collection and ad adn remove operations are thread safe Ref: https://learn.microsoft.com/en-us/dotnet/csharp/language-reference/language-specification/classes

Reply:
Added parallel ATS for 25 ATS calls in the sample App (tests/devapps/WAM/NetFrameworkWam/Form1.cs). Still works.